src/main/java/io/sprucehill/zalando/api/nativecart/service/AbstractService.java

```diff
@@ -0,0 +1,285 @@
+package io.sprucehill.zalando.api.nativecart.service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.sprucehill.zalando.api.model.Domain;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.*;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * @author Michael Duergner
+ */
+public abstract class AbstractService {
```

Comment:
I am not really imagine the changes as of now that is required in this class, so copied it for now , when we have a testable version of APIs i will merge/work on this class.